### PR TITLE
my92xx integration (second round)

### PR DESCRIPTION
**Pull request by lougovsk**
Reworked the my92xx driver based on the adv_pwm driver and integrated into the 5.0.14

### CODE REVIEW

**Comment by lougovsk** [comment]
@RavenSystem can you please take a look

**Comment by lougovsk** [comment]
Is it stale? :)

**Comment by lougovsk** [comment]
Well, that’s kinda mean

File: devices/HAA/header.h [hunk 1]
```diff
@@ -254,6 +254,7 @@
 #define TH_SENSOR_ERROR_COUNT               ch_group->num[9]
 
 #define LIGHTBULB_TYPE                      ch_group->num[5]
+#define LIGHTBULB_DRIVER_TYPE_SET           "dr"
```
Comment (RavenSystem) [inline devices/HAA/header.h:257]: Not needed. LIGHTBULB_TYPE must be used to avoid waste of RAM.

File: devices/HAA/main.c [hunk 2]
```diff
@@ -7483,6 +7484,12 @@ void normal_mode_init() {
         lightbulb_group->armed_autodimmer = false;
         lightbulb_group->autodimmer_task_delay = AUTODIMMER_TASK_DELAY_DEFAULT;
         lightbulb_group->autodimmer_task_step = AUTODIMMER_TASK_STEP_DEFAULT;
+        lightbulb_group->funcs.start = adv_pwm_start;
+        lightbulb_group->funcs.stop = adv_pwm_stop;
+        lightbulb_group->funcs.set_freq = adv_pwm_set_freq;
+        lightbulb_group->funcs.set_duty = adv_pwm_set_duty;
+        lightbulb_group->funcs.get_duty = adv_pwm_get_duty;
+        lightbulb_group->funcs.new_channel = adv_pwm_new_channel;
```
Comment (RavenSystem) [inline devices/HAA/main.c:7492]: Those declarations are not needed. Waste of RAM. LIGHTBULB_TYPE must be used.

File: libs/my92xx/lightbulb_my92xx_driver.c [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+void my92xx_send();
+
+static my92xx_cmd_t config = {
+    .bit_width = MY92XX_CMD_BIT_WIDTH_16,
+    .scatter = MY92XX_CMD_SCATTER_APDM,
+    .frequency = MY92XX_CMD_FREQUENCY_DIVIDE_1,
+    .reaction = MY92XX_CMD_REACTION_FAST,
+    .one_shot = MY92XX_CMD_ONE_SHOT_DISABLE,
+    .resv = 0
+};
```
Comment (RavenSystem) [inline libs/my92xx/lightbulb_my92xx_driver.c:60]: Not needed data struct. Waste of RAM.

File: libs/my92xx/lightbulb_my92xx_driver.c [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+static my92xx_driver_info_t info = {
+    .di_pin = 13,
+    .dcki_pin = 15,
+    .numChips = 1,
+    .numChannels = 0,
+    .channels = NULL
+};
```
Comment (RavenSystem) [inline libs/my92xx/lightbulb_my92xx_driver.c:67]: More unnecessary data.

**Comment by RavenSystem** [comment]
A complete rework is necessary. I will do it when I have enough free time.

File: libs/my92xx/lightbulb_my92xx_driver.c (hunk 3 above)
Comment (lougovsk) [reply libs/my92xx/lightbulb_my92xx_driver.c:60]: Do you suggest to allocate it dynamically or rather that one can avoid use of the struct and directly make byte array?

File: devices/HAA/main.c (hunk 2 above)
Comment (lougovsk) [reply devices/HAA/main.c:7492]: Do you suggest for each adv_pwm function call check the type of bulb and either call pwn or my92xx driver?

**Pull request closed without merging**